chatbot/chatbot_agent.py

```diff
@@ -0,0 +1,318 @@
+## Import Python Packages
+import os
+import requests
+from typing import List
+
+import openai
+import chromadb
+import langchain
+
+from langchain.embeddings.openai import OpenAIEmbeddings
+from langchain.vectorstores import Chroma
+from langchain.text_splitter import TokenTextSplitter
+from langchain.llms import OpenAI
+from langchain.chains import ChatVectorDBChain
+from langchain.chains import RetrievalQA
+
+from langchain.document_loaders import GutenbergLoader
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseLoader
+
+from langchain.prompts import PromptTemplate
+from langchain.chains.qa_with_sources import load_qa_with_sources_chain
+from langchain.chains.question_answering import load_qa_chain
+
+
+
+
+## chatbot agent class
+class ChatbotAgent:
+    def __init__(self, _openai_api_key: str, _sources_urls: List[str], _persist_directory: str):
+        """
+        Initializes an instance of the ChatbotAgent class.
+
+        Args:
+            openai_api_key (str): OpenAI API key.
+            sources_urls (List[str]): URLs of the files to be merged from Project Open-academy.
+            persist_directory (str): Directory where the Chroma vectors will be persisted.
+        """
+        # Set OpenAI API key
+        self.__openai_api_key = _openai_api_key
+        os.environ["OPENAI_API_KEY"] = self.__openai_api_key
+        self.sources_urls = _sources_urls

```

Comment:
Please keep the code style consistent. Should you have _ as prefix for self.sources_urls as well?

Reply:
Done